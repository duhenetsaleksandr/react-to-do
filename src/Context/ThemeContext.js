import React from 'react';

const ThemeContext = React.createContext({ darkTheme: true, switchTheme: () => {} });

export default ThemeContext;
