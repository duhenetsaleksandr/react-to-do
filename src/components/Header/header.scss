@use '../../styles/colors';

.header {
    background-color: colors.$primary-color;
    color: colors.$secondary-color;

    &__container {
        height: 50px;
        display: flex;
        align-items: center;
        justify-content: space-between;
    }
}
