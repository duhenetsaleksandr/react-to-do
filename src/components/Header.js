import React from 'react';
import '../styles/header.scss';
import Logo from './Logo';

export default function Header() {
    return (
        <header className="header">
            <div className="container header__container">
                <Logo/>
            </div>
        </header>
    );
}
