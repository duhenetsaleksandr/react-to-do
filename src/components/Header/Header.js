import React, { useContext } from 'react';
import ThemeContext from 'Context/ThemeContext';
import './Header.scss';
import Logo from 'components/common/Logo';
import SwitcherTheme from 'components/common/SwitcherTheme';
import SwitcherLang from 'components/common/SwitcherLang';

const Header = ({ currentLang, onChangeLocale }) => {
    const { darkTheme } = useContext(ThemeContext);
    return (
        <header className={`header ${darkTheme ? 'dark' : 'light'}`}>
            <div className='container header__container'>
                <Logo/>
                <SwitcherLang currentLang={currentLang} onChangeLocale={onChangeLocale}/>
                <SwitcherTheme/>
            </div>
        </header>
    );
};

export default Header;
