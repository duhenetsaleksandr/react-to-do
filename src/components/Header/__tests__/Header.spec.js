import React from 'react';
import Header from '../Header';
import ThemeContext from 'Context/ThemeContext';

describe('should render Header component', () => {
    let component;
    beforeEach(() => {
        component = mount(<Header/>);
    });

    it('should contain tag header with class .header', () => {
        const header = component.find('header.header');
        expect(header).toHaveLength(1);
    });

    it('should contain div with class .container and .header__container', () => {
        const header = component.find('div.container.header__container');
        expect(header).toHaveLength(1);
    });

    it('should contain component Logo', () => {
        const logo = component.find('Logo');
        expect(logo).toHaveLength(1);
    });

    it('should render Header with light theme using context', () => {
        const component = mount(<Header/>, {
            wrappingComponent: ThemeContext.Provider,
            wrappingComponentProps: {
                value: { darkTheme: false },
            },
        });

        const footer = component.find('header.header.light');
        expect(footer).toHaveLength(1);
    });

    it('should render Header with dark theme using context', () => {
        const component = mount(<Header/>, {
            wrappingComponent: ThemeContext.Provider,
            wrappingComponentProps: {
                value: { darkTheme: true },
            },
        });

        const footer = component.find('header.header.dark');
        expect(footer).toHaveLength(1);
    });

    it('should render Header component snapshot', () => {
        const component = shallow(<Header/>);
        expect(component).toMatchSnapshot();
    });
});
