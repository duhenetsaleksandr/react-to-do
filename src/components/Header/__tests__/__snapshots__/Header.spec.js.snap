// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component should render Header component snapshot 1`] = `
<header
  className="header dark"
>
  <div
    className="container header__container"
  >
    <Logo />
    <SwitcherLang />
    <SwitcherTheme />
  </div>
</header>
`;
